### DISCUSSION TITLE
Can affine transformation backpropagate gradient?

### DISCUSSION THREAD

**Discussion by JiajiZhu01** [TorchIO-project/torchio#1034] [Q&A]
Hi, we are using torchio as part of our code to train a 3D MRI registration model. The model output is the 9 affine transformation parameters (scales,degrees,translations). We will then use these parameters to transforme the 3D MRI, get the central slice in the Z direction and compare it to a groundtruth 2D image to get the loss. However, tio.affine cannot backpropage losses. Is there any other way to do this?

**Comment by romainVala** [comment]
Hi

no the affine transfo is base on sitk, which do the all computation with numpy so no backpropagation.

You want to learn a coregistration ? this is something I would like to test too, but I did not find the time to do it.
if this is the case, may be you should have a closer look at 

https://github.com/Hsankesara/VoxelMorph-PyTorch
or
https://github.com/daviddmc/SVoRT

I hope it helps, and if you find out other solutions, let me know

Romain
